src/date.c

```diff
@@ -1,18 +1,1006 @@
 /*
```

Comment:
Optional, we can first focus on functional aspects. 
Once we improved the quality of the code by working together on this, and we can
fix the style later.

One thing though: I noticed that your files have trailing whitespaces. Removing whitespaces is a good practice. `git diff` even marks them red.  But the most important thing is that if somebody (e.g. I) have an editor on save hook that removes them on save, it will basically introduce a diff in a dozen places of the `date.c` file on the first save, which is annoying.